### PR TITLE
Add relevance features to 2020-06-30 and 2020-06-30-preview

### PR DESCRIPTION
**Pull request by shmed**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 1]
```diff
@@ -275,6 +275,33 @@
...
           },
+          {
+            "name": "scoringStatistics",
+            "in": "query",
+            "type": "string",
+            "enum": [
+              "local",
+              "global"
+            ],
+            "x-ms-enum": {
```
Comment (brjohnstmsft) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:286]: We've recently added explicit enum label names and descriptions to all `x-ms-enum` declarations. Please follow the same pattern here; There should be lots of examples in the specs now.

Also, if we think it's at all likely we might add new values besides `local` and `global` in the future, we should use `modelAsString: true` here.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 2]
```diff
@@ -275,6 +275,33 @@
...
+              "name": "ScoringStatistics",
+              "modelAsString": false
+            },
+            "x-nullable": false,
+            "description": "A value that specifies whether we want to calculate scoring statistics (such as document frequency) globally for more consistent scoring, or locally, for lower latency.",
+            "x-ms-parameter-grouping": {
+              "name": "SearchParameters"
```
Comment (brjohnstmsft) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:293]: The parameter group is called `SearchOptions` in Track 2

**Review by bitsforthought — CHANGES_REQUESTED** [review]
Enum and parameter group declarations need some changes

**Comment by brjohnstmsft** [comment]
The breaking changes CI failures are expected, since all new parameters to Search POST become new response properties for the continuation token. However, as we discussed internally when designing these features, we do not consider ranking changes to be behaviorally breaking changes. Since these new parameters do not materially affect the response other than ranking, we consider this a backward compatible change.

We'll need an exception to get this merged once my PR feedback is addressed.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 1 above)
Comment (shmed) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:286]: I dont think there's any plan on adding new values to this enum.

**Review by bitsforthought — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

**Pull request merged**